feat: added SecureTcpIp to eControlMethod enum for plugin.

jkdevito opened the pull request:
Actual SecureTcpIp client is not currently available.

Closes #121

andrew-welker commented on Pepperdash Core/Pepperdash Core/Comm/GenericSecureTcpIpClient.cs:
```diff
@@ -0,0 +1,1048 @@
+        /// <summary>
+        /// Internal secure client
+        /// </summary>
+        public SecureTCPClient Client { get; private set; }
```
This should be private and a field.
```
private SecureTCPClient _client;
```

As we're wrapping the client and adding additional functionality, we don't want it to be accessible from outside of this class.

jkdevito replied:
Changed client to private and renamed to reflect a private instance.

andrew-welker commented on Pepperdash Core/Pepperdash Core/Comm/GenericSecureTcpIpClient.cs:
```diff
@@ -0,0 +1,1048 @@
+        /// <summary>
+        /// Band aid delegate for choked server
+        /// </summary>
+        internal delegate void ConnectionHasHungCallbackDelegate();
```
This can probably be removed. I'm guessing it was put in the original class to allow for handling Crestron's SecureTcpServer connections

jkdevito replied:
Removed unused delegate.

jkdevito commented:
Added GenericSecureTcpIpClient, based off existing GenericSecureTcpIpClient_ForServer class, updated to implement ISocketStatusWithStreamDebugging.  New client is currently untested.
